### PR TITLE
EventGrid: Regenerate the SDK for the new 2018-09-15-preview API version and update tests and recorded sessions.

### PR DESCRIPTION
**Pull request by ahamad-MS**
## Description

- Regenerated the SDK for the new 2018-09-15-preview API version.

- Add new test cases and update existing test cases to cover the new features, namely, domains/domaintopics and advanced filtering.

* Re-recorded the tests.

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [X] Please add REST spec PR link to the SDK PR:  
https://github.com/Azure/azure-rest-api-specs/pull/3765 

- [X] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [X] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [X] Title of the pull request is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [X] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [X] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [X] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [X] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Review by dsgouda — CHANGES_REQUESTED** [review]
Left a few comments

File: src/SDKs/EventGrid/management/Management.EventGrid/Microsoft.Azure.Management.EventGrid.csproj [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
-		<PackageTags>Microsoft Azure EventGrid Management;Event Grid;Event Grid management;</PackageTags>
-		<PackageReleaseNotes>This is a preview SDK for the new features introduced in 2018-05-01-preview API version. The stable version of the SDK targeting the 2018-01-01 API version continues to exist as version 1.3.0</PackageReleaseNotes>
-	</PropertyGroup>
-	<PropertyGroup>
-		<TargetFrameworks>net452;netstandard1.4</TargetFrameworks>
-	</PropertyGroup>
+  <PropertyGroup>
+    <PackageId>Microsoft.Azure.Management.EventGrid</PackageId>
+    <Description>Provides developers with a library to create and manage all Azure EventGrid resources. </Description>
+    <Version>3.0.0-preview</Version>
```
[the remark this answers is not in the document]
Comment (ahamad-MS) [reply src/SDKs/EventGrid/management/Management.EventGrid/Microsoft.Azure.Management.EventGrid.csproj]: We already have a stable version 1.3.0 of the SDK which is targeting the GA'ed 2018-01-01 API version. The plan for the new GA version is still under discussion in the team.

File: src/SDKs/EventGrid/Management.EventGrid/Generated/SdkInfo_EventGridManagementClient.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  internal static partial class SdkInfo
+  {
+      public static IEnumerable<Tuple<string, string, string>> ApiInfo_EventGridManagementClient
+      {
+          get
+          {
+              return new Tuple<string, string, string>[]
+              {
+                new Tuple<string, string, string>("EventGrid", "DomainTopics", "2018-09-15-preview"),
```
[the remark this answers is not in the document]
Comment (ahamad-MS) [reply src/SDKs/EventGrid/Management.EventGrid/Generated/SdkInfo_EventGridManagementClient.cs]: @dsgouda  Addressed. thx

File: src/SDKs/EventGrid/management/Management.EventGrid/Microsoft.Azure.Management.EventGrid.csproj (hunk 1 above)
[the remark this answers is not in the document]
Comment (ahamad-MS) [reply src/SDKs/EventGrid/management/Management.EventGrid/Microsoft.Azure.Management.EventGrid.csproj]: @dsgouda All comments are addressed and all checks are passed. please let me know if more info needed. thanks.

**Review by dsgouda — APPROVED** [review]
LGTM

**Pull request merged**